### PR TITLE
Modbus endian-ness & OctectStream @type

### PR DESCRIPTION
**Pull request by fillobotto**
- Modbus: try to guess endian-ness of received buffer from the passed `contentType`  parameter `byteSeq`
- Modbus: tell to the server which endianness to use when reading a buffer by looking at the same property as above
- Core: OctetStream serializer/deserializer to prioritize `@type` over `type` property
- Core: added OctetStream tests

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -1,19 +1,30 @@
 {
     // See https://go.microsoft.com/fwlink/?LinkId=733558
     // for the documentation about the tasks.json format
-    "version": "0.1.0",
+    "version": "2.0.0",
```
Comment (relu91) [inline .vscode/tasks.json]: May I ask you to remove this committed file?

File: packages/binding-modbus/src/modbus-client.ts [hunk 2]
```diff
@@ -110,6 +112,22 @@ export default class ModbusClient implements ProtocolClient {
...
 
+  private validateEndianness(form: ModbusForm) : ModbusEndianness {
+    let endianness = ModbusEndianness.BIG_ENDIAN
+    if(form.contentType) {
+      const contentValues : string[] = form.contentType.split(';') ?? []
+      // Check endian-ness
+      const byteSeq = contentValues.find(value => /^byteSeq=/.test(value))
+      if(byteSeq) {
+        const guessEndianness = byteSeq.split('=')[1] as ModbusEndianness
+        if(guessEndianness) {
```
Comment (relu91) [inline packages/binding-modbus/src/modbus-client.ts:123]: What if guessEndianness is not among the admitted values? the function seems to just extract the right value, but it does not a proper validation

File: packages/binding-modbus/test/modbus-client-test.ts [hunk 3]
```diff
@@ -329,6 +329,62 @@ describe('Modbus client test', () => {
...
+                href: "modbus://127.0.0.1:8502",
+                contentType: "application/octet-stream;length=2;byteSeq=LITTLE_ENDIAN_BYTE_SWAP",
+                "modbus:function": 16,
+                "modbus:offset": 0,
+                "modbus:unitID": 1
+            }
+
+            await client.writeResource(form, { type: "", body: Buffer.from([ 0x59, 0x60, 0x25, 0x49 ]) })
+            testServer.registers.should.be.deep.equal([9545, 22880], "wrong coil value")
+        });
```
Comment (relu91) [inline packages/binding-modbus/test/modbus-client-test.ts:421]: What about reading? might it be reduntant?

File: packages/core/src/codecs/octetstream-codec.ts [hunk 4]
```diff
@@ -106,6 +128,7 @@ export default class OctetstreamCodec implements ContentCodec {
                         return bigendian ? bytes.readFloatBE(0) : bytes.readFloatLE(0);
 
                     case 8:
+                        console.error("BOIAAIIAAI")
```
Comment (relu91) [inline packages/core/src/codecs/octetstream-codec.ts]: 🤣 delete ?

File: packages/core/src/codecs/octetstream-codec.ts [hunk 5]
```diff
@@ -57,8 +59,28 @@ export default class OctetstreamCodec implements ContentCodec {
...
+        // check @type property for further information
+        if(schema['@type'] !== undefined) {
+            let semTypes: Array<string> = schema["@type"];
+            // check for endianness semantic type
+            // see http://www.meta-share.org/ontologies/meta-share/meta-share-ontology.owl/documentation/index-en.html
+            let endianSem = semTypes.find(v => v.endsWith(":bigEndian") || v.endsWith(':littleEndian'));
+            bigendian = endianSem === undefined ? bigendian : endianSem.endsWith(":bigEndian");
+            // check for numeric datatype semantic
+            // see https://www.w3.org/TR/xmlschema-2/
+            let typeSem = semTypes.find(v => /:(unsigned)?(int|long|double|float)/.test(v.toLowerCase()));
```
Comment (relu91) [inline packages/core/src/codecs/octetstream-codec.ts]: short and byte?

**Review by relu91 — CHANGES_REQUESTED** [review]
Just minor comments/questions plus some minor changes.

File: packages/binding-modbus/src/modbus-client.ts (hunk 2 above)
Comment (fillobotto) [reply packages/binding-modbus/src/modbus-client.ts:123]: Should I throw an error or just discard the parameter (falling back to Big Endian)? @relu91

**Comment by fillobotto** [comment]
Something weird happened with the rebase (?) even tough the signature seems working now

File: packages/binding-modbus/test/modbus-client-test.ts (hunk 3 above)
Comment (fillobotto) [reply packages/binding-modbus/test/modbus-client-test.ts:421]: It comes out that test are needed so I added 2 read tests. They should be enough

**Review by relu91 — CHANGES_REQUESTED** [review]
Looks better now but there is still the task.json file as changed, could you remove it?

**Comment by relu91** [comment]
> Something weird happened with the rebase (?) even tough the signature seems working now

Yep it does not look super good, I'll take care of it, maybe I'll just cherry-pick the commits 👍🏻 . Don't worry about that for now

**Comment by relu91** [comment]
Manually merged in e980eddc63c8146e9f8fa63b2a361cffe7adaf12 . There are still unsigned commits we may need to fix them. First let me create the PR for the main fork.

**Pull request closed without merging**